framework/src/meshgenerators/BlockDeletionGenerator.C

```diff
@@ -22,18 +22,27 @@ BlockDeletionGenerator::validParams()
 
   params.addClassDescription(
       "Mesh generator which removes elements with the specified subdomain ID");
-  params.addRequiredParam<SubdomainID>("block_id", "The block to be deleted");
+  params.addParam<std::vector<SubdomainID>>("block_ids", "The list of blocks to be deleted");
+  params.addDeprecatedParam<SubdomainID>(
+      "block_id", 0, "The block to be deleted. Use block_ids instead");
 
   return params;
 }
 
 BlockDeletionGenerator::BlockDeletionGenerator(const InputParameters & parameters)
-  : ElementDeletionGeneratorBase(parameters), _block_id(getParam<SubdomainID>("block_id"))
+  : ElementDeletionGeneratorBase(parameters),
+    _block_ids(getParam<std::vector<SubdomainID>>("block_ids"))
 {
+  // Handle deprecated parameter
+  if (!parameters.isParamSetByUser("block_id"))
+    _block_ids.push_back(getParam<SubdomainID>("block_id"));
+  if (!parameters.isParamSetByUser("block_id") && !parameters.isParamSetByUser("block_ids"))
+    paramError("block_ids",
```

Comment:
This won't work. The param has to exist for paramError to be useful. Use `mooseError`